[Docs][Alerting] updates images in Alerts Management and adds docs for the Run When field (#92225)
elastic/kibana: docs/user/alerting/defining-alerts.asciidoc

[role="screenshot"]
image::images/alert-flyout-action-type-selection.png[UI for selecting an action type]

> When an alert instance is detected as matching a condition it is marked as _Active_, assigned an action group and triggers the actions in that group.

Reviewer: ```suggestion
When an alert instance matches a condition, the alert is marked as _Active_ and assigned an action group.  The actions in that group are triggered.
```

> When the condition is no longer detected for an _Active_ alert instance, it is assigned the _Recovered_ action group, triggering any actions assigned to that group.

Reviewer: ```suggestion
When the condition is no longer detected, the alert is assigned to the _Recovered_ action group, which triggers any actions assigned to that group.
```

> **Run When** allows you to assign an action to an _action group_, ensuring you are notified about alert instances in that group by triggering the action in accordance with your **Notify every** setting.

Reviewer: ```suggestion
**Run When** allows you to assign an action to an _action group_. This will trigger the action in accordance with your **Notify every** setting.
```

> Each action must specify a <<alerting-concepts-connectors, connector>> instance. If no connectors exist for that action type, click "Add new" to create one.

Reviewer: ```suggestion
Each action must specify a <<alerting-concepts-connectors, connector>> instance. If no connectors exist for that action type, click *Add new* to create one.
```

Reviewer: Is the button label "Add new" or "Add action"?

Author: Good catch! That was the existing text 😬 
I've updated it.

Thanks @gchaps !
